python_modules/libraries/dagster-celery/dagster_celery/tasks.py

```diff
@@ -214,8 +217,99 @@ def _execute_step_k8s_job(
     return _execute_step_k8s_job
 
 
+def create_docker_task(celery_app, **task_kwargs):
+    @celery_app.task(bind=True, name='execute_step_docker', **task_kwargs)
+    def _execute_step_docker(
+        _self,
+        instance_ref_dict,
+        step_keys,
+        run_config,
+        mode,
+        repo_name,
+        repo_location_name,
+        run_id,
+        docker_image,
+        docker_creds
+    ):
+        '''Run step execution in a Docker container.
+        '''
+        import docker.client
+        from dagster_graphql.client.mutations import (
+            handle_execution_errors,
+            handle_execute_plan_result,
+        )
+
+        instance_ref = InstanceRef.from_dict(instance_ref_dict)
+        instance = DagsterInstance.from_ref(instance_ref)
+        pipeline_run = instance.get_run_by_id(run_id)
+        check.invariant(pipeline_run, 'Could not load run {}'.format(run_id))
+
+        step_keys_str = ", ".join(step_keys)
+
+        variables = {
+            'executionParams': {
+                'runConfigData': run_config,
+                'mode': mode,
+                'selector': {
+                    'repositoryLocationName': repo_location_name,
+                    'repositoryName': repo_name,
+                    'pipelineName': pipeline_run.pipeline_name,
+                },
+                'executionMetadata': {'runId': run_id},
+                'stepKeys': step_keys,
+            }
+        }
+
+        command = 'dagster-graphql -v \'{variables}\' -p executePlan'.format(
+            variables=seven.json.dumps(variables)
+        )
+
+        client = docker.client.from_env()
+        client.login(**docker_creds)
+
+        # Post event for starting execution
+        engine_event = instance.report_engine_event(
+            'Executing steps {} in Docker container {}'.format(step_keys_str, docker_image),
+            pipeline_run,
+            EngineEventData(
+                [
+                    EventMetadataEntry.text(step_keys_str, 'Step keys'),
+                ],
+                marker_end=DELEGATE_MARKER,
+            ),
+            CeleryDockerEngine,
+            step_key=step_keys[0],
+        )
+
+        events = [engine_event]
+
+        docker_response = client.containers.run(
+            docker_image,
+            command=command,
+            detach=False,
+            auto_remove=True,
+            # pass through this worker's environment for things like AWS creds etc.
+            environment=dict(os.environ),
+        )
+
+        try:
+            res = seven.json.loads(docker_response)
+        except:
+            print(docker_response)
```

Comment:
we will probably want to report an event to the instance here, and possibly fail the run

Reply:
what errors do we expect here? I've found a strange hack in seven.json
`
try:
    from json import JSONDecodeError
except ImportError:
    JSONDecodeError = ValueError
`
is it for python 2.7 ?
would seven.json.JSONDecodeError be enough to handle?

could you please point me an example of explicit failing a run?